File: Sources/Indigox.SSO.Application/js/Indigox/SSO/Application/LogRecord.js
define("Indigox/SSO/Application/LogRecord",
    [
        "Indigox.Web.JsLib.Models.RecordManager"
    ],
function (
    RecordManager
) {
    RecordManager.getInstance().register('LogRecord', {
        columns: [
            { name: 'ID', text: '编号', type: String },
            { name: 'AccountName', text: '用户名', type: String },
            { name: 'IP', text: '登录IP', type: String },
            { name: 'LogTime', text: '登录时间', type: String },
            { name: 'Action', text: '类别', type: String }
        ],
        primaryKey: ['ID']
    });

});

File: Sources/Indigox.SSO/Logs/LogRecord.cs
using System;

namespace Indigox.SSO.Logs
{
    public class LogRecord
    {
        private int id;
        private string accountName;
        private string ip;
        private DateTime logTime;
        private ActionType action;

        public LogRecord() { }

        public LogRecord(string accountName, string ip, ActionType action)
        {
            this.accountName = accountName;
            this.ip = ip;
            this.action = action;
            this.logTime = DateTime.Now;
        }

        public int ID 
        { 
            get { return this.id; }
        }

        public string AccountName 
        { 
            get { return this.accountName; } 
        }

        public string IP 
        { 
            get { return this.ip; } 
        }

        public DateTime LogTime 
        { 
            get { return this.logTime; } 
        }

        public ActionType Action 
        { 
            get { return this.action; } 
        }
    }

    public enum ActionType
    {
        Login,Logout
    }
}
